packages/components-library/src/components/ui/Pagination.vue

```diff
@@ -0,0 +1,287 @@
+
+      this.val = { ...this.value, ...{ loading: false, count, page } }
+    },
+    navigate (page) {
+      if (!this.useRouter) { return this.fetchData(page) }
+
+      if (this.$route.query.page) {
+        if (
+          parseInt(this.$route.query.page, 10) !== page ||
+          parseInt(this.$route.query.size, 10) !== this.value.size
+        ) {
+          // (!) Merge with existing query; e.g. bookmarks
+          this.$router.push({
+            path: this.$route.path,
+            query: { ...this.$route.query, page, size: this.value.size }
+          })
+        }
+      } else {
+        this.$router.replace({
+          path: this.$route.path,
+          query: { ...this.$route.query, page, size: this.value.size }
+        })
+      }
+    },
+    updateValue (value) {
+      this.val = { ...this.val, ...this.value, ...value }
+      this.$emit('input', { ...this.value, ...this.val, ...value })
+    }
+  },
+  props: {
+    /**
+     * Add css classes to pagination buttons.
+     */
+    styles: {
+      type: Object,
+      required: false,
+      default: () => ({})
+    },
+    /**
+     * The method to retrieve items with.
+     * @returns {Object} {count: Number}
+     */
+    method: {
+      type: Function,
+      required: false
+    },
+    /**
+     * Amount of navigational page buttons around the current page button.
+     */
+    midPageSize: {
+      type: Number,
+      default: () => 4
+    },
+    /**
+     * Whether to integrate with Vue-router.
+     */
+    useRouter: {
+      type: Boolean,
+      default: () => true
+    },
+    /**
+     * The internal pagination model
+     * @model
+     */
+    value: {
+      type: Object,
+      required: true
+    }
+  },
+  watch: {
+    '$route.query': {
+      handler: function (query) {
+        if (!this.method || !this.useRouter) { return }
+
+        if (query.page) {
+          this.fetchData(parseInt(query.page, 10))
+        } else {
+          this.navigate(1)
+        }
+      },
+      deep: true,
+      immediate: true
+    },
+    value: {
+      handler: function (newVal) {
+        this.val = newVal
+      },
+      deep: true
+    },
+    'value.size' () {
+      if (!this.method) return
+      // Changing the amount of items per page results
+      // in going back to page 1 and recalculation of
+      // the paginator.
+      this.navigate(1)
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+.c-pagination {
+  width: 100%;
+
+  .button-container {
+    display: flex;
+
+    .btn {
+      // Use fixed width for buttons, so their position
+      // remains stable while navigation
+      width: 2.5rem;
+
+      &.ellipsis {
+        margin: 0 0.5rem;
+        width: auto;
+      }
+    }
+
+    .pagination-controls {
+      select {
+        width: 4rem;
+      }
+    }
+  }
+
+  .item-count {
+    font-weight: 600;
+  }
+}
+</style>
+
+<docs>
+## Basic example
+
+```jsx
+  let model = { size: 20, page: 1 }
```

Comment:
How to i know the shape of the model as a user ( maybe add typescript for model) and or describe as part of docs tag

Reply:
Added a comment above.